### ISSUE TITLE
Binary Timeseries Output Format

### ISSUE THREAD

**Issue by aaron-boranian** [NREL/EnergyPlus#6589]
Right now EnergyPlus outputs the hourly/timestep variables and meters to eso and/or sqlite. Then, the eso can be translated into csv. Both have some issues:

eso -> csv is buggy and the csv cannot always be read. Additionally there’s an inherent loss in precision when serializing to text-based formats.

The other option, SQLite, solves some of those problems, but can be really slow to read large output files.

I propose having an option to output in HDF5, which is a cross-platform binary tabular data format used extensively in high performance computing applications. I/O is very fast and there is no loss of precision. It could help both EnergyPlus to save run time on I/O as well as users processing results.

Migrated from [UserVoice](http://energyplus.uservoice.com/forums/258860/suggestions/16139431) feedback from @nmerket

**Comment by nmerket** [comment]
It seems like this has stalled out a bit. @jasondegraw brought up that there are issues with the HDF5 library. 

I'd like to throw out the idea of outputting to the [Apache Parquet](https://parquet.apache.org/) format instead. I'm doing work with large batches of EnergyPlus simulations and analyzing the time series results is becoming increasingly important and more of a bottleneck on file i/o from CSV or SQLite. Parquet is a "native" format for Hadoop/Spark and makes the kinds of analysis we'd like to do much more performant. It appears [there is an existing C++ library](https://github.com/apache/parquet-cpp) with a [permissive license](https://github.com/apache/parquet-cpp/blob/master/LICENSE.txt) we could use for this purpose.

Again, this wouldn't replace the CSV output, but would be another option.

**Comment by jasondegraw** [comment]
@nmerket I didn't look too hard so maybe I'm missing something, but it looks like boost is one of the dependencies of the C++ library.

**Comment by nmerket** [comment]
I haven't looked very deeply into it either, but it does appear that way. Is Boost going to be a problem?

**Comment by jasondegraw** [comment]
Yes, it is. Will the output changes in #6192 do anything to help your bottleneck?

**Comment by nmerket** [comment]
Even if it's part of a third party library? It seems like we're okay with it on a project level already. 

Msgpack is interesting. That might work. It just seems like an overly generic format when there are others (like parquet) that are more geared towards tabular numeric data.

**Comment by jasondegraw** [comment]
The small piece we have will hopefully be removed in the future.

I don't know much about CBOR or MessagePack, but it does superficially seem like it could help with your application. There are also possible solutions with an expanded API, let's make sure that output is part of those discussions going forward.

**Comment by rHorsey** [comment]
Additional binary support will be instrumental for various 100k+ scale simulation sets we'll be engaging in over the next few years. MessagePack would help substantially - I agree that Parquet would make more sense, but either way I'd love movement on this issue.

Thanks!
